src/Inventory/Asset/NetworkPort.php

```diff
@@ -299,7 +316,35 @@ private function handleMacConnection(\stdClass $port, int $netports_id)
             if (isset($this->connections['NetworkEquipment'])) {
                 return;
             }
-            $this->handleHub($found_macs, $netports_id);
+
+            // see if we got a mix of different device types (other than computers)
+            if (!(isset($this->connection_ports['Computer']) && count($this->connection_ports) == 1)) {
+                $this->handleHub($found_macs, $netports_id);
+                return;
+            }
+
+            $item_ids = [];
+            $real_port_ids = [];
+            foreach (array_keys($found_macs) as $k) {
+                $networkPort = new \NetworkPort();
+                $networkPort->getFromDB($k);
```

Comment:
```suggestion
                if (!$networkPort->getFromDB($k)) {
                    continue;
                }
```

Reply:
This is redundant: `$found_macs` is an array of NetworkPort keys.